pkg/controller/cloud/node_controller.go

```diff
@@ -147,6 +147,14 @@ func (cnc *CloudNodeController) updateNodeAddress(node *v1.Node, instances cloud
 		glog.V(5).Infof("This node %s is still tainted. Will not process.", node.Name)
 		return
 	}
+	// Node that isn't present according to the cloud provider shouldn't have its address updated
+	exists, err := ensureNodeExistsByProviderIDOrExternalID(instances, node)
+	if err != nil {
+		glog.Errorf("%v", err)
+	} else if !exists {
+		glog.V(2).Infof("The node %s is no longer present according to the cloud provider, do not process.", node.Name)
```

Comment:
This will log for every node that isn't managed by the cloud provider, correct? Thoughts on making this `V(4)` to reduce spam in those cases?

Reply:
Yes, I fixed it, please PTLA again.  @jhorwit2